### PR TITLE
fix: (IntegrationContext) add process instance execution context attributes

### PR DESCRIPTION
**Pull request by igdianov**
This PR injects process instance execution context attributes into IntegrationContext holder class at runtime.

### CODE REVIEW

File: activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java [hunk 1]
```diff
@@ -54,7 +56,29 @@ private IntegrationContextImpl buildFromExecution(DelegateExecution execution,
...
+                integrationContext.setProcessDefinitionKey(processInstance.getProcessDefinitionKey());
+                integrationContext.setProcessDefinitionVersion(processInstance.getProcessDefinitionVersion());
+                
+                // Let's try extract parent process instance from super execution  
+                if(processInstance.getSuperExecutionId() != null) {
+                    try {
+                        Optional.ofNullable(processInstance.getSuperExecution())
+                                .ifPresent(superExecution -> integrationContext
+                                .setParentProcessInstanceId(superExecution.getProcessInstanceId()));
+                    } catch (NullPointerException e) {
```
[the remark this answers is not in the document]
Comment (erdemedeiros) [reply activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java]: @igdianov I think it makes sense to ignore it in this case. I'm just wondering if can avoid this catch NPE. Maybe by updating the if block to `if(processInstance.getSuperExecutionId() != null && Context.getCommandContext() != null ) {`?

[the remark this answers is not in the document]
Comment (igdianov) [reply activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java]: @erdemedeiros I was thinking that if IntegrationContextBuilder intended use is only inside Activiti transaction internally, we should catch that NPE when trying to extract parent process Id and fail by throwing an ActivitiException to signal the problem. Ignoring NPE does not seem right to me at this moment.

[the remark this answers is not in the document]
Comment (erdemedeiros) [reply activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java]: @igdianov as far as I know this code will always be called from inside an Activiti transaction as it's used by service tasks behaviours. The question is how capital the the information about parent process id is. Shaw we fail if we are not able to retrieve it or shaw we just ignore it?
But I see your point, maybe it's safer to throw an error so we know if this code get invoked outside of an Activiti transaction in the future.

[the remark this answers is not in the document]
Comment (igdianov) [reply activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java]: @erdemedeiros It seems to me that parent process id is important part of the IntegrationContext contract. We should be able to to resolve and provide it to connectors if it exists. If we cannot resolve superExecution due to improper usage outside of engine context then it is better to fail fast. If you agree, I can make a test for this case.

[the remark this answers is not in the document]
Comment (erdemedeiros) [reply activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java]: @igdianov yes, it makes sense. I'm ok with this approach.

**Pull request merged**